doc/source/ray-core/ray-dag.rst

```diff
@@ -0,0 +1,155 @@
+executed as root node while iterating, or used as input args or kwargs of other
+functions to form more complex DAGs.
+
+.. tabbed:: Python
+
+    .. code-block:: python
+
+        # `ray start --head` has been run to launch a local cluster
+        import ray
+
+        @ray.remote
+        def func(src, inc=1):
+            return src + inc
+
+        a_ref = func.bind(1, inc=2)
+        assert ray.get(a_ref.execute()) == 3 # 1 + 2 = 3
+        b_ref = func.bind(a_ref, inc=3)
+        assert ray.get(b_ref.execute()) == 6 # (1 + 2) + 3 = 6
+        c_ref = func.bind(b_ref, inc=a_ref)
+        assert ray.get(c_ref.execute()) == 9 # ((1 + 2) + 3) + (1 + 2) = 9
+
+
+Ray DAG with classes and class methods
+--------------------------------------
+
+The IR node generated by ``.bind()`` on a ``ray.remote`` decorated class is
+executed as a Ray Actor upon execution. The Actor will be instantiated every
+time the node is executed, and the classmethod calls can form a chain of
+function calls specific to the parent Actor instance.
+
+DAG IR nodes generated from a function, class or classmethod can be combined
+together to form a DAG.
+
+.. tabbed:: Python
+
+    .. code-block:: python
+
+        # `ray start --head` has been run to launch a local cluster
+        import ray
+
+        @ray.remote
+        class Actor:
+            def __init__(self, init_value):
+                self.i = init_value
+
+            def inc(self, x):
+                self.i += x
+
+            def get(self):
+                return self.i
+
+        a1 = Actor.bind(10)  # Instantiate Actor with init_value 10.
+        val = a1.get.bind()  # ClassMethod that returns value from get() from
+                             # the actor created.
+        assert ray.get(val.execute()) == 10
+
+        @ray.remote
+        def combine(x, y):
+            return x + y
+
+        a2 = Actor.bind(10) # Instantiate another Actor with init_value 10.
+        a1.inc.bind(2)  # Call inc() on the actor created with increment of 2.
+        a1.inc.bind(4)  # Call inc() on the actor created with increment of 4.
+        a2.inc.bind(6)  # Call inc() on the actor created with increment of 6.
+        # Combine outputs from a1.get() and a2.get()
+        dag = combine.bind(a1.get.bind(), a2.get.bind())
+
+        # a1 +  a2 + inc(2) + inc(4) + inc(6)
+        # 10 + (10 + ( 2   +    4    +   6)) = 32
+        assert ray.get(dag.execute()) == 32
+
+
+Ray DAG with custom InputNode
+-----------------------------
+
+``InputNode`` is the singleton node of a DAG that represents user input value at
+runtime. It should be used within a context manager with no args, and called
+as args of ``dag_node.execute()``
+
+.. tabbed:: Python
+
+    .. code-block:: python
+
+        # `ray start --head` has been run to launch a local cluster
+        import ray
+        # TODO (jiaodong): Update this once DAG is out of experimental folder.
+        from ray.experimental.dag.input_node import InputNode
```

Comment:
it will hard to remember this, any chance to put these snippets to a test file and use literal_include?

Reply:
soon this will just be `from ray.dag import InputNode` tho, it might be explanatory ?